update login, register, and profile template pages and navigation bar

phoebebai opened the pull request:
Finish theme login, register and profile template pages.

AustinGrey commented on web_server/social_distribution/templates/index.html:
```diff
@@ -3,25 +3,59 @@
         integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
+	<script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js"
+		integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo"
+		crossorigin="anonymous"></script>
```
Okay, I'm seeing some dependencies here. 

- jQuery
- Bootstrap
- Popper

We can't use the requirements.txt to document these since that's for python module dependencies only. But we should still document them. Do you mind adding these to the README?

AustinGrey commented on web_server/social_distribution/templates/index.html:
```diff
@@ -3,25 +3,59 @@
-
-    {% block content %}{% endblock content %}
-
-    <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js"
```
I might be out to lunch here, but if the scripts are moved to the head rather than the foot of the document, then if they aren't protected with a onready function call and they attempt to change the document on load, the document might not have rendered yet and the js function might fail. It's just something to be aware of.

AustinGrey commented on web_server/users/static/users/css/mandala.css:
```diff
@@ -0,0 +1,167 @@
+/*
+
+Copyright (c) 2020 - YIFEI BAI - https://codepen.io/soju22/pen/gEYJoO
```
This copyright notice is confusing. You have stated that you have the copyright, but then link to codepen.io, then the license notice is MIT. Our entire project is by default MIT licensed, and I dont think copyright is something we can claim off a codepen. I would remove the license notice, and replace it with an author statement (e.g. @Author: Yifei Bai) and then provide the codepen link. To be polite, it might also be of value to also state the author of the original codepen.

AustinGrey commented on web_server/users/templates/users/login.html:
```diff
@@ -1,26 +1,51 @@
+	crossorigin="anonymous"></script>
+<link rel="stylesheet" href="{% static '/users/css/style.css' %}">
+<script src='https://cdn.jsdelivr.net/npm/vue'></script>
+<script src='https://cdnjs.cloudflare.com/ajax/libs/d3/5.9.1/d3.min.js'></script>
```
I'm also seeing Vue and D3 as dependencies here. Are these just for the mandala animation on this page?

AustinGrey commented on web_server/users/static/users/css/style.css:
```diff
@@ -0,0 +1,73 @@
+	text-align: center;
+}
+
+.profile-part-I {
```
These class names are hard to guess what they are for. I would look at the page but due to a database issue I cannot get this branch to run. These classes should get named to what they mean (e.g. profile-icon, or profile-content) or they could be separated out so that the outer .profile block has an inner .icon block and the rule could be changed to ".profile > .icon"

AustinGrey commented on web_server/users/static/users/js/mandala.js:
```diff
@@ -0,0 +1,263 @@
+const palette = [
```
I assume this file and the mandala.css file are all part of that code pen? I would suggest saying so in a comment as well on both these files. We might get hit with plagiarism if we don't

AustinGrey commented on web_server/users/templates/users/mandala.html:
```diff
@@ -0,0 +1,352 @@
+
+      <script src='https://cdn.jsdelivr.net/npm/vue'></script>
+      <script src='https://cdnjs.cloudflare.com/ajax/libs/d3/5.9.1/d3.min.js'></script>
+      <link rel="stylesheet" href="{% static '/users/css/mandala.css' %}">
```
Okay so I'm also seeing the vue and d3 dependencies duplicated here. Are we using Vue and D3 in places other than this mandala script? If not we might be leaking dependencies to other pages which could slow load times for other pages unnecessarily.

AustinGrey commented:
@XuechunHou just reminded me that the reason I can't log in with the newly created account is that it is not active. I went and activated the account and was able to log in fine. However I still get the first error when attempting to edit fields.

phoebebai commented:
I added code for profile function in users/views.py. It should work now

XuechunHou commented:
1. logout page doesn't work  (like "Log in Again is not clickable anymore")
![image](https://user-images.githubusercontent.com/33268064/75215854-4df29d80-574f-11ea-9ded-a132c55bc5e1.png)
2. log in page is not responsive.  text gets cut off when browser window size is small
![image](https://user-images.githubusercontent.com/33268064/75216013-cce7d600-574f-11ea-9cc0-e8222501093e.png)
3. profile page could show a success message after successfully edited.

XuechunHou commented:
I see in register.html, bootstrap, jquery and popper are loaded again (they are already loaded in index.html). I am thinking all the dependencies should be defined in index.html and nowhere else. and all other pages extend index.html so that they have access to the external libraries.. ?

I also see there is a nav bar thing added in index.html, I am thinking can we move it to individual pages and leave the body of index.html empty, and all other pages extend index.html...

AustinGrey commented:
@XuechunHou regarding the text getting cut off when the window is small. Targeting both small form factor and large form factor devices seems unnecessary for this assignment. Since we only have 2 front end developers and as far as I am aware neither has extensive grid, flex-box, or even media query experience the amount of work to make a responsive design is far greater than simply twice as much effort. Can we define a minimum window size of 720p and say graphical issues do not matter under this size?

Regarding 3. this might be a good enhancement that could be added as an issue to the project board so it doesn't block this PR but can still be done later.

phoebebai commented:
@XuechunHou
1,The logout page is the django's built page. I am not sure why it does not work, since it works here. Therefore, I changed it to redirect to login page. Please take a look. Hopefully it works.
2, Since we need to implement at least 1/2 of the requirements by 13 March, I will fix it later if I still have time.
3, I added problem 3 as an issue to the project board, I will work on it later tomorrow.

XuechunHou commented:
I made a change that the main nav bar is now decoupled from index.html and is put in a new file called "navigation.html". Also, all pages are now extending either navigation.html (which extends index.html) when the page does need the main nav bar displayed, or index.html directly when the page does not need the main nav bar.
